homeassistant/components/alexa/smart_home.py

```diff
@@ -799,35 +928,140 @@ def __init__(self, smart_home_config):
         return b'' if response is None else self.json(response)
 
 
+class _AlexaDirective:
+    def __init__(self, request):
+        self._directive = request[API_DIRECTIVE]
+        self.namespace = self._directive[API_HEADER]['namespace']
+        self.name = self._directive[API_HEADER]['name']
+        self.payload = self._directive[API_PAYLOAD]
+        self.has_endpoint = API_ENDPOINT in self._directive
```

Comment:
So for 1 device being multiple entities in Alexa, we should be able to solve that with our new Device Registry too. If integrations are configured using config entries, they can now register to what device they belong.